packages/grid/x-data-grid/src/models/colDef/gridColType.ts

```diff
@@ -11,3 +11,26 @@ export type GridNativeColTypes =
 
 // Use `LiteralUnion` to get autocompletion for literal types.
 export type GridColType = LiteralUnion<GridNativeColTypes, string>;
+
+export type GridNativeColOperator =
+  | 'is'
+  | 'not'
+  | 'after'
+  | 'onOrAfter'
+  | 'before'
+  | 'onOrBefore'
+  | 'isEmpty'
+  | 'isNotEmpty'
+  | '='
+  | '!='
+  | '>'
+  | '>='
+  | '<'
+  | '<='
+  | 'isAnyOf'
+  | 'contains'
+  | 'equals'
+  | 'startsWith'
+  | 'endsWith';
```

Comment:
Can you explain how you generated this list? It will help me review & ensure we don't miss anything.

Also, do you see any way to generate this automatically? It is likely to become stale otherwise.

Reply:
@romgrk  - I generated the list by going thru each of the operators files in `colDef` (eg `gridDateOperators.ts`) and compiling the `value` property. Agreed that it would be better to dynamically generate this list. One way could be creating an enum with all of the operator names, and then referencing it in the operators files. Then we could create this new `GridNativeColOperator` type similar to this example: https://stackoverflow.com/a/55143029

Reply:
Sounds like a safer way to do it, do you want to implement that?